src/utils/httpRegionUtils.ts

```diff
@@ -1,4 +1,5 @@
 import * as models from '../models';
+import { resetDependentRegions } from '../plugins/core/metaData/refMetaDataHandler';
```

Comment:
This line is probably the reason for the faulty tests. I haven't quite understood the reason yet, but it is triggered by a cyclic dependency. Cause is that jest seems to load the files in a different way (vm.runInContext?) than by normal execution. However, in doing so, cyclic dependencies are not loaded correctly and null/ undefined access can occur.
This is a big reason why I recently restructured the project. I've since been using a rule about which imports are allowed in which folders: `plugins` -> `store` -> `utils` -> `io` -> `models`. 
`utils` -> `plugins` cause the cyclic dependency.

Reply:
I instead moved the functions to `utils/httpRegionUtils` as it makes more sense to have them there. All in all that solution simply looks cleaner.